cognite/client/data_classes/workflows.py

```diff
@@ -1249,6 +1281,51 @@ def _load_trigger(cls, data: dict) -> WorkflowScheduledTriggerRule:
         return cls(cron_expression=data.get("cronExpression"))
 
 
+class WorkflowDataModelingTriggerRule(WorkflowTriggerRule):
+    """
+    This class represents a data modeling trigger rule.
+
+    Args:
+        data_modeling_query (WorkflowTriggerDataModelingQuery): The data modeling query of the trigger.
+        batch_size (int | None): The batch size of the trigger.
+        batch_timeout (int | None): The batch timeout of the trigger.
+    """
+
+    _trigger_type = "dataModeling"
+
+    def __init__(
+        self,
+        data_modeling_query: WorkflowTriggerDataModelingQuery,
+        batch_size: int | None = None,
+        batch_timeout: int | None = None,
+    ) -> None:
+        super().__init__()
+        self.data_modeling_query = data_modeling_query
+        self.batch_size = batch_size
+        self.batch_timeout = batch_timeout
+
+    @classmethod
+    def _load_trigger(cls, data: dict) -> WorkflowDataModelingTriggerRule:
+        return cls(
+            data_modeling_query=WorkflowTriggerDataModelingQuery.load(typing.cast(dict, data.get("dataModelingQuery"))),
```

Comment:
Either the `"dataModelingQuery"` key is optional, and `WorkflowTriggerDataModelingQuery` needs to accept `dict | None`, or it is not optional, and you should not use `.get` (either way, no `typing.cast` should be here)

```suggestion
            data_modeling_query=WorkflowTriggerDataModelingQuery.load(data["dataModelingQuery"])),
```

Reply:
fixed